src/network/networkd-ndisc.c

```diff
@@ -1635,38 +2070,70 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return r;
 
-        r = ndisc_drop_outdated(link, timestamp_usec);
+        r = ndisc_drop_outdated(link, /* router = */ NULL, timestamp_usec);
         if (r < 0)
                 return r;
 
-        r = ndisc_start_dhcp6_client(link, rt);
+        /* 3. Update the default router. */
+        r = ndisc_remember_default_router(link, rt);
         if (r < 0)
                 return r;
+        is_default = r;
+
+        /* 4. Process the RA header.
+         * Here, to gracefully support the case that there exist multiple routers with different preferences
+         * and different settings, the header elements are only processed when the router has the highest
+         * preference. Otherwise, e.g. when two routers send different hop limit, the sysctl setting
+         * frequently flipped. Of course, this is not perfect, as the two routers may have the same
+         * preference. */
+        if (is_default) {
+                /* 4.1. Start DHCPv6 client based on the flag. */
+                r = ndisc_start_dhcp6_client(link, rt);
+                if (r < 0)
+                        return r;
 
-        r = ndisc_router_process_default(link, rt);
-        if (r < 0)
-                return r;
+                /* 4.2. Apply reachable time. */
+                r = ndisc_router_process_reachable_time(link, rt);
```

Comment:
This implementation of Reachable Time is not compliant with IPv6 certification tests.  Please see https://github.com/systemd/systemd/issues/31842.